### ISSUE TITLE
Implement inline submitting elements on page/parent save

### ISSUE THREAD

**Issue by emteknetnz** [dnadesign/silverstripe-elemental#1215]
This card is for the implementation of "Approach one" from [SPIKE Look at solutions between field validation conflicting between block level and page level](https://github.com/silverstripe/silverstripe-elemental/issues/1204)

## Issues this resolves
- https://github.com/silverstripe/silverstripe-elemental/issues/1179
- https://github.com/silverstripe/silverstripe-elemental/issues/1177

## Questions:
- Do we want green toast notifications for individual content blocks published when page is published? (e.g. you'll get three green toasts on page save if two elements are saved)
- Do we want red toast notifications for individual content blocks published when they fail validation? (currently there are no red toasts)
- Do we want to block saving individual elements until all elements are valid? (Would be significantly more complex, need validation endpoints)

## Acceptance criteria
- [ ] Elements with a dirty form state are inline saved when the page/parent save/publish button is clicked
- [ ] The page/parent save/publish is blocked when there are validation errors with one or more of the elements that were inline saved
- [ ] Solution works with 
  - [ ] ElementalAreas on Pages 
  - [ ] ElementalAreas on non-page DataObjects
  - [ ] Multiple ElementalAreas on one form
- [ ] Elements are still cascade published on parent publish
- [ ] PHP logic for handling saving elemental data on page/parent save is no longer called and formally deprecated.
- [ ] If viable, stop submitting elemental data on page/parent save with POST request.
 
## Note
- We've thought about adding an opt-out for the new behaviour but decide against it under the assumption people would have had to customise the individual block save anyway.
- Our testing team has offered to help regression test the recent elemental work prior to the beta. Follow up with them once this card is done.

## Shared CI run (elemental endtoend) are the relevant jobs)
- https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/10071368437

Notes:
- The elemental endtoend job is the relevant one
- I cannot replicate any of the other endtoend failures locally
- The phpunit failures are unrelated

## PRs
- https://github.com/silverstripe/silverstripe-frameworktest/pull/193
- https://github.com/silverstripe/silverstripe-behat-extension/pull/277
- https://github.com/silverstripe/silverstripe-framework/pull/11301
- https://github.com/silverstripe/silverstripe-admin/pull/1792
- https://github.com/silverstripe/silverstripe-elemental/pull/1214
- https://github.com/silverstripe/developer-docs/pull/542

## Test PRs
- https://github.com/silverstripe/silverstripe-admin/pull/1796

**Comment by GuySartorelli** [comment]
PRs merged.

There's a note in the issue description that says

> Our testing team has offered to help regression test the recent elemental work prior to the beta. Follow up with them once this card is done.

Reassigning to Steve to do that.

**Comment by emteknetnz** [comment]
Have contacted testing team regrading this, will close this issue now that the PRs have been merged. We'll deal with any regressions found by the testing team in a separate issue
